auslib/admin/views/base.py

```diff
@@ -20,39 +20,20 @@ def decorated(*args, **kwargs):
     return decorated
 
 
-def requirepermission(url, options=['product']):
-    def wrap(f):
-        def decorated(*args, **kwargs):
-            username = request.environ.get('REMOTE_USER', request.environ.get("HTTP_REMOTE_USER"))
-            method = request.method
-            extra = dict()
-            for opt in options:
-                if opt in request.form:
-                    extra[opt] = request.form[opt]
-                elif request.get_json() and opt in request.json:
-                    extra[opt] = request.json[opt]
-                else:
-                    msg = "Couldn't find required option %s in form" % opt
-                    logging.warning("Bad input: %s", msg)
-                    return Response(status=400, response=msg)
-            if not dbo.permissions.hasUrlPermission(username, url, method, urlOptions=extra):
-                msg = "%s is not allowed to access %s by %s" % (username, url, method)
-                logging.warning("Unauthorized access attempt: %s", msg)
-                return Response(status=401, response=msg)
-            return f(*args, **kwargs)
-        return decorated
-    return wrap
-
-
-def catchOutdatedDataError(messages):
+def handleGeneralExceptions(messages):
     def wrap(f):
         def decorated(*args, **kwargs):
+            # FIXME: Need to rollback the transaction if 4xx errors happen.
```

Comment:
Further evidence that this is fine: I created a situation locally where a PermissionDeniedError was raised after an update was done, and I added a log message as part of AUSTransaction.rollback. Output was as follows:
```
balrogadmin_1 | [pid: 210|app: 0|req: 2/2] 172.17.42.1 () {46 vars in 805 bytes} [Tue May 31 12:51:36 2016] GET /api/csrf_token => generated 20 bytes in 2 msecs (HTTP/1.1 200) 8 headers in 432 bytes (1 switches on core 0)
balrogadmin_1 | 2016-05-31 12:51:36,302 - DEBUG - PID: 210 - Request: 140232923627344 - SingleReleaseView.put#55: processing PUT request to /releases/B2G-gecko-nightly-latest
balrogadmin_1 | 2016-05-31 12:51:36,304 - DEBUG - PID: 210 - Request: 140232923627344 - Releases.getReleases#940: Looking for releases with:
balrogadmin_1 | 2016-05-31 12:51:36,304 - DEBUG - PID: 210 - Request: 140232923627344 - Releases.getReleases#941: name: B2G-gecko-nightly-latest
balrogadmin_1 | 2016-05-31 12:51:36,304 - DEBUG - PID: 210 - Request: 140232923627344 - Releases.getReleases#942: product: None
balrogadmin_1 | 2016-05-31 12:51:36,308 - DEBUG - PID: 210 - Request: 140232923627344 - AUSTransaction.execute#108: Attempting to execute SELECT releases.`read_only` 
balrogadmin_1 | FROM releases 
balrogadmin_1 | WHERE releases.name = %s
balrogadmin_1 | 2016-05-31 12:51:36,309 - DEBUG - PID: 210 - Request: 140232923627344 - AUSTransaction.execute#108: Attempting to execute SELECT releases.product 
balrogadmin_1 | FROM releases 
balrogadmin_1 | WHERE releases.name = %s
balrogadmin_1 | 2016-05-31 12:51:36,309 - DEBUG - PID: 210 - Request: 140232923627344 - auslib.blobs.apprelease.ReleaseBlobV4.validate#70: Validating blob {u'platforms': {u'aries-userdebug': {u'locales': {u'en-US': {u'buildID': u'20151112022457', u'platformVersion': u'45.0a1', u'displayVersion': u'45.0a1', u'completes': [{u'fileUrl': u'https://queue.taskcluster.net/v1/task/IlV-0N-DQhC3PBkctAQtlg/runs/0/artifacts/public/build/b2g-aries-gecko-update.mar', u'from': u'*', u'hashValue': u'94b9b0564418d610853e3b1f422ef08d4efbdc9e5362d606f7617228b007c0aa74009cdc44f7dbf2c0a87c40e15d015a0cb47270a22151901bf3b0cc79e4e1d6', u'filesize': 131059753}], u'appVersion': u'45.0a1'}}}, u'flame-userdebug': {u'locales': {u'en-US': {u'isOSUpdate': True, u'buildID': u'20151110120524', u'appVersion': u'45.0a1', u'displayVersion': u'45.0a1', u'platformVersion': u'45.0a1', u'completes': [{u'fileUrl': u'https://queue.taskcluster.net/v1/task/V1HyfEHJRPG3va_W1MfIJQ/runs/0/artifacts/public/build/fota-flame-update.mar', u'from': u'*', u'hashValue': u'2685288aa4896685e4aa611d0c1be83a06641f585aa11e26a81444707bc8a6efdd65bebb2d6cf9f31a14e56839f416c1361fef2059f1a7ff33a0fd7974a7f634', u'filesize': 108017496}]}}}, u'niiiiiiexus': {u'locales': {u'en-US': {u'isOSUpdate': True, u'buildID': u'20151110113826', u'appVersion': u'44.0a2', u'displayVersion': u'44.0a2', u'platformVersion': u'44.0a2', u'completes': [{u'fileUrl': u'https://queue.taskcluster.net/v1/task/WBQBJAhDSTik2zkk_LY_2A/runs/0/artifacts/public/build/fota-hammerhead-update.mar', u'from': u'*', u'hashValue': u'8b796b1a6240f56e958b236b82a0cfff80aee3c604aae4a4cac7b7e733c7123ee719b7be2d6e855a6958a4ea6fd4306b8e79d322a3f9aba0c6723e01c21715d3', u'filesize': 101589914}]}}}, u'flame': {u'locales': {u'en-US': {u'isOSUpdate': True, u'buildID': u'20151110120640', u'appVersion': u'45.0a1', u'displayVersion': u'45.0a1', u'platformVersion': u'45.0a1', u'completes': [{u'fileUrl': u'https://queue.taskcluster.net/v1/task/U7LxpmZvTUqo2-UqMXYGRQ/runs/0/artifacts/public/build/fota-flame-update.mar', u'from': u'*', u'hashValue': u'1d67ea90413c3a8cc6017dda1210ed90afebbd4525678bd52f1f6c6900cd7c27cf04b16c648c9e4340fa688c00ba1f1a1092e803c8a29b441817fa1a70c709fb', u'filesize': 107923517}]}}}, u'aries': {u'locales': {u'en-US': {u'buildID': u'20151130145625', u'platformVersion': u'45.0a1', u'displayVersion': u'45.0a1', u'appVersion': u'45.0a1', u'completes': [{u'fileUrl': u'https://queue.taskcluster.net/v1/task/Do8_llXIRXCZ6uao3hfVvQ/runs/0/artifacts/public/build/b2g-aries-gecko-update.mar', u'from': u'*', u'hashValue': u'd094e5a7a912aac5fca107958aaf55924114902bc7a33f3cb3056752df760287ba3ee11582366e7880a50278ca4406ab7a1394bdf741e73fa5ba6a18f3b39e4a', u'filesize': 132127809}]}}}}, u'hashFunction': u'sha512', u'name': u'B2G-gecko-nightly-latest', u'schema_version': 4}
balrogadmin_1 | 2016-05-31 12:51:36,353 - DEBUG - PID: 210 - Request: 140232923627344 - AUSTransaction.execute#108: Attempting to execute SELECT releases.name, releases.product, releases.`read_only`, releases.data, releases.data_version 
balrogadmin_1 | FROM releases 
balrogadmin_1 | WHERE releases.name = %s
balrogadmin_1 | 2016-05-31 12:51:36,354 - DEBUG - PID: 210 - Request: 140232923627344 - AUSTransaction.execute#108: Attempting to execute UPDATE releases SET name=%s, product=%s, `read_only`=%s, data=%s, data_version=%s WHERE releases.name = %s AND releases.data_version = %s
balrogadmin_1 | 2016-05-31 12:51:36,355 - DEBUG - PID: 210 - Request: 140232923627344 - AUSTransaction.execute#108: Attempting to execute INSERT INTO releases_history (changed_by, timestamp, name, product, `read_only`, data, data_version) VALUES (%s, %s, %s, %s, %s, %s, %s)
balrogadmin_1 | 2016-05-31 12:51:36,356 - DEBUG - PID: 210 - Request: 140232923627344 - AUSTransaction.__exit__#87: exc is:
balrogadmin_1 | Traceback (most recent call last):
balrogadmin_1 |   File "./auslib/admin/views/base.py", line 57, in put
balrogadmin_1 |     return self._put(*args, transaction=trans, **kwargs)
balrogadmin_1 |   File "./auslib/admin/views/base.py", line 19, in decorated
balrogadmin_1 |     return f(*args, changed_by=username, **kwargs)
balrogadmin_1 |   File "./auslib/admin/views/releases.py", line 236, in _put
balrogadmin_1 |     old_data_version=data_version, transaction=transaction)
balrogadmin_1 |   File "./auslib/db.py", line 1093, in updateRelease
balrogadmin_1 |     raise PermissionDeniedError("wtf!")
balrogadmin_1 | PermissionDeniedError: wtf!
balrogadmin_1 | 2016-05-31 12:51:36,356 - DEBUG - PID: 210 - Request: 140232923627344 - AUSTransaction.rollback#131: Rolling back

```

...and a manual look at the release showed that its contents were unchanged.